crates/cranelift/src/func_environ.rs

```diff
@@ -2174,6 +2233,22 @@ impl<'module_environment> cranelift_wasm::FuncEnvironment for FuncEnvironment<'m
         if self.tunables.epoch_interruption {
             self.epoch_function_entry(builder);
         }
+
+        let func_index = match &builder.func.name {
+            UserFuncName::User(user) => {
+                FuncIndex::from_u32(user.index)
+            }
+            _ => {
+                unreachable!()
+            }
+        };
+        let func_name = self.translation.debuginfo.name_section.func_names[&func_index];
```

Comment:
Have we tested what happens if we don't have a name section? We should I think handle this case gracefully (I guess by treating it as if no functions match). You can test this by doing `wasm-strip program.wasm` and then running (`wasm-strip` is part of Binaryen I think; happy to help or run this for you if needed).